pkg/cgroup/resolve_container.go

```diff
@@ -90,16 +120,21 @@ func GetContainerInfo(cGroupID, pid uint64, withCGroupID bool) (*ContainerInfo,
 		return info, err
 	}
 
-	if i, ok := containerIDToContainerInfo[containerID]; ok {
+	instance.mu.Lock()
+	defer instance.mu.Unlock()
```

Comment:
```suggestion
	instance.mu.Lock()
	defer instance.mu.Unlock()
```
We should let the callee handle Mutex than the caller.

Reply:
I'm sorry I don't understand the leaky abstraction here. These are the functions in the same file that are accessing the cache. How does one avoid the locking here? Or is the suggestion to have more functions for the cache structure in terms of setters and getters that these functions call essentially making them wrappers around these setters and getters?